pms-115-120/utility/src/main/java/io/mosip/pms/ida/service/PMSDataMigrationService.java

```diff
@@ -55,46 +64,94 @@ public class PMSDataMigrationService {
 
 	@Autowired
 	MispLicenseRepository mispLicenseRepository;
+	
+	@Autowired
+	LastSyncRepository lastSyncRepository;
 
 	@Autowired
 	private Environment environment;
 
 	@Autowired
 	private ObjectMapper mapper;
+	
+	@Autowired
+	private CertUtil certUtil;
+	
+	@Autowired
+	RestUtil restUtil;
 
 	@Autowired
 	private WebSubPublisher webSubPublisher;
 
 	public void initialize() {
-		LOGGER.error("Started publishing the data");
+    	LocalDateTime lastSync = getLastSyncTimeStamp();
+    	LocalDateTime latestSync = LocalDateTime.now();
+		LOGGER.info("Started publishing the data");
 		try {
-			publishAPIKeyData();
-			publishMISPLicenseData();
+				publishPartnerUpdated(lastSync,latestSync);
+				publishAPIKeyData(lastSync,latestSync);
+				publishMISPLicenseData(lastSync,latestSync);
+				if(lastSync!=null) {
+					publishUpdateApiKey(lastSync,latestSync);
+				}
 		} catch (Exception e) {
-			LOGGER.error("Error occurred while publishing the data");
+			LOGGER.error("Error occurred while publishing the data - " + e);
 		}
+		saveLatestSyncTimeStamp(latestSync);
```

Comment:
I think this should be inside the try block, so if there is any issue that timestamp is not saved

Reply:
okay sir, I'll update the same as suggested